php/src/Docker/DockerHubManager.php

```diff
@@ -26,31 +26,29 @@ public function GetLatestDigestOfTag(string $name, string $tag) : ?string {
         // If one of the links below should ever become outdated, we can still upgrade the mastercontainer via the webinterface manually by opening '/api/docker/getwatchtower'
 
         try {
-            $request = $this->guzzleClient->request(
+            $authTokenRequest = $this->guzzleClient->request(
                 'GET',
-                'https://hub.docker.com/v2/repositories/' . $name . '/tags?page_size=128'
+                'https://auth.docker.io/token?service=registry.docker.io&scope=repository:' . $name . ':pull'
             );
-            $body = $request->getBody()->getContents();
+            $body = $authTokenRequest->getBody()->getContents();
             $decodedBody = json_decode($body, true);
-
-            if (isset($decodedBody['results'])) {
-                $arch = php_uname('m') === 'x86_64' ? 'amd64' : 'arm64';
-                foreach($decodedBody['results'] as $values) {
-                    if (isset($values['name'])
-                    && $values['name'] === $tag
-                    && isset($values['images'])
-                    && is_array($values['images'])) {
-                        foreach ($values['images'] as $images) {
-                            if (isset($images['architecture'])
-                            && $images['architecture'] === $arch
-                            && isset($images['digest'])
-                            && is_string($images['digest'])) {
-                                $latestVersion = $images['digest'];
-                                apcu_add($cacheKey, $latestVersion, 600);
-                                return $latestVersion;
-                            }
-                        }
-                    }
+            if(isset($decodedBody['token'])) {
+                $authToken = $decodedBody['token'];
+                $manifestRequest = $this->guzzleClient->request(
+                    'GET',
+                    'https://registry-1.docker.io/v2/'.$name.'/manifests/' . $tag,
+                    [
+                        'headers' => [
+                            'Accept' => 'application/vnd.docker.distribution.manifest.v2+json,application/vnd.oci.image.index.v1+json',
```

Comment:
@Zoey2936 are you saying multiarch images use `application/vnd.docker.distribution.manifest.list.v2+json`?
Edit: looks like it...

Reply:
However this is actually a problem since when not a new image is returned, it willl not show that an update is available...

So maybe we should change this now and build multiarch images later when this update is out?

Reply:
> @Zoey2936 are you saying multiarch images use `application/vnd.docker.distribution.manifest.list.v2+json`? Edit: looks like it...

yes:
application/vnd.docker.distribution.manifest.list.v2+json for multiarch
application/vnd.docker.distribution.manifest.v2+json for single arch

Reply:
I see. Then we have a problem as stated above...
As if we push this out now, no one will see that an update is available because it is not able to get the new manifest...

So what about doing this change now and coming back to this later when the update is already out for a long time?

Reply:
No, since:
```
curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.docker.distribution.manifest.v2+json"  https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/latest -Is | grep docker-content-digest
docker-content-digest: sha256:92e71fcc71eba6f9f1b64b981e3241693dc7398c6009cff2fe23ab3bb44d5121
```
The digest is wrong, but there is a digest, so it will show as outdated

Reply:
```suggestion
                            'Accept' => 'application/vnd.oci.image.index.v1+json,application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json',
```

Reply:
Are you sure this will not return the latest non-multiarch manifest which might be the one that you are currently using?

Reply:
In my test, there was never a single arch image pushed to latest, and it still sends a response

Reply:
Still i'd like to test this. Will try to do so tomorw.

Reply:
```Dockerfile
FROM alpine
```
```console
root@mx:~/t# curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.oci.image.index.v1+json,application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/latest -Is | grep docker-content-digest
root@mx:~/t# docker build --provenance false -t zoeyvid/alpine:latest --push .
[+] Building 7.9s (6/6) FINISHED                                                                                                                                                                                             
 => [internal] load .dockerignore                                                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                                                         0.0s
 => [internal] load build definition from Dockerfile                                                                                                                                                                    0.0s
 => => transferring dockerfile: 49B                                                                                                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                        0.0s
 => CACHED [1/1] FROM docker.io/library/alpine                                                                                                                                                                          0.0s
 => exporting to image                                                                                                                                                                                                  0.0s
 => => exporting layers                                                                                                                                                                                                 0.0s
 => => writing image sha256:25b55c980e48f4a21a2b7ecd8c859436b29b3c9c2bdd7958dbcf43bb7596a1a5                                                                                                                            0.0s
 => => naming to docker.io/zoeyvid/alpine:latest                                                                                                                                                                        0.0s
 => pushing zoeyvid/alpine:latest with docker                                                                                                                                                                           5.7s
 => => pushing layer 8e012198eea1                                                                                                                                                                                       5.6s
root@mx:~/t# curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.oci.image.index.v1+json,application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/latest -Is | grep docker-content-digest
docker-content-digest: sha256:fa3d9cace9122740a3ced3810cc237d6d6be72e9036396829d86f3a7f25161cf
root@mx:~/t# curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/latest -Is | grep docker-content-digest
docker-content-digest: sha256:fa3d9cace9122740a3ced3810cc237d6d6be72e9036396829d86f3a7f25161cf
root@mx:~/t# docker buildx build --provenance false -t zoeyvid/alpine:arm64 --push . --platform arm64
[+] Building 0.0s (0/0)                                                                                                                                                                                                      
[+] Building 2.8s (8/8) FINISHED                                                                                                                                                                                             
 => [internal] load build definition from Dockerfile                                                                                                                                                                    0.0s
 => => transferring dockerfile: 49B                                                                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                        1.2s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                                                                           0.0s
 => [1/1] FROM docker.io/library/alpine@sha256:f271e74b17ced29b915d351685fd4644785c6d1559dd1f2d4189a5e851ef753a                                                                                                         0.0s
 => => resolve docker.io/library/alpine@sha256:f271e74b17ced29b915d351685fd4644785c6d1559dd1f2d4189a5e851ef753a                                                                                                         0.0s
 => exporting to image                                                                                                                                                                                                  1.5s
 => => exporting layers                                                                                                                                                                                                 0.0s
 => => exporting manifest sha256:456f196cd6a39d0b3d06ff0026b0d91297ce0ac8e1ff945d8cf40583bb06d15f                                                                                                                       0.0s
 => => exporting config sha256:6f3f17645df550391d690aea482f54731a140c63b685a947d8288f42557d72a3                                                                                                                         0.0s
 => => pushing layers                                                                                                                                                                                                   1.2s
 => => pushing manifest for docker.io/zoeyvid/alpine:arm64@sha256:456f196cd6a39d0b3d06ff0026b0d91297ce0ac8e1ff945d8cf40583bb06d15f                                                                                      0.3s
 => [auth] zoeyvid/alpine:pull,push token for registry-1.docker.io                                                                                                                                                      0.0s
 => [auth] library/alpine:pull zoeyvid/alpine:pull,push token for registry-1.docker.io                                                                                                                                  0.0s
root@mx:~/t# docker buildx build --provenance false -t zoeyvid/alpine:amd64 --push . --platform amd64
[+] Building 2.4s (5/5) FINISHED                                                                                                                                                                                             
 => [internal] load build definition from Dockerfile                                                                                                                                                                    0.0s
 => => transferring dockerfile: 49B                                                                                                                                                                                     0.0s
 => [internal] load .dockerignore                                                                                                                                                                                       0.0s
 => => transferring context: 2B                                                                                                                                                                                         0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                        1.0s
 => [1/1] FROM docker.io/library/alpine@sha256:f271e74b17ced29b915d351685fd4644785c6d1559dd1f2d4189a5e851ef753a                                                                                                         0.0s
 => => resolve docker.io/library/alpine@sha256:f271e74b17ced29b915d351685fd4644785c6d1559dd1f2d4189a5e851ef753a                                                                                                         0.0s
 => exporting to image                                                                                                                                                                                                  1.4s
 => => exporting layers                                                                                                                                                                                                 0.0s
 => => exporting manifest sha256:92e71fcc71eba6f9f1b64b981e3241693dc7398c6009cff2fe23ab3bb44d5121                                                                                                                       0.0s
 => => exporting config sha256:0ba2b9314db6130620130d7f7c6361d86a13573a4551a7c55a6aeda890424136                                                                                                                         0.0s
 => => pushing layers                                                                                                                                                                                                   1.1s
 => => pushing manifest for docker.io/zoeyvid/alpine:amd64@sha256:92e71fcc71eba6f9f1b64b981e3241693dc7398c6009cff2fe23ab3bb44d5121                                                                                      0.3s
root@mx:~/t# docker manifest create zoeyvid/alpine:latest -a zoeyvid/alpine:amd64 -a zoeyvid/alpine:arm64
Created manifest list docker.io/zoeyvid/alpine:latest
root@mx:~/t# docker manifest push zoeyvid/alpine:latest
sha256:41a7beafbb7018eb23b88eecf40cbf9caf04d05591f24fc8e29c32b9ae8cfc9f
root@mx:~/t# curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/latest -Is | grep docker-content-digest
docker-content-digest: sha256:92e71fcc71eba6f9f1b64b981e3241693dc7398c6009cff2fe23ab3bb44d5121
root@mx:~/t# curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.oci.image.index.v1+json,application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/latest -Is | grep docker-content-digest
docker-content-digest: sha256:41a7beafbb7018eb23b88eecf40cbf9caf04d05591f24fc8e29c32b9ae8cfc9f
root@mx:~/t# curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.oci.image.index.v1+json,application/vnd.docker.distribution.manifest.list.v2+json,application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/arm64 -Is | grep docker-content-digest
docker-content-digest: sha256:456f196cd6a39d0b3d06ff0026b0d91297ce0ac8e1ff945d8cf40583bb06d15f
root@mx:~/t# curl -H "Authorization: Bearer $(curl "https://auth.docker.io/token?service=registry.docker.io&scope=repository:zoeyvid/alpine:pull" -s | jq -r .token)" -H "Accept: application/vnd.docker.distribution.manifest.v2+json" https://registry-1.docker.io/v2/zoeyvid/alpine/manifests/arm64 -Is | grep docker-content-digest
docker-content-digest: sha256:456f196cd6a39d0b3d06ff0026b0d91297ce0ac8e1ff945d8cf40583bb06d15f
```

Reply:
Okay looks good to me. Thanks a lot for testing!